core/services/src/main/java/org/exoplatform/services/rest/AttachmentsRestService.java

```diff
@@ -192,13 +206,26 @@ public Response DeleteEntityAttachment(@ApiParam(value = "Entity technical ident
       return Response.status(Response.Status.BAD_REQUEST).entity("Entity type is mandatory").build();
     }
 
-    try {
-      attachmentService.deleteAttachmentItemById(entityId, entityType, attachmentId);
-      return Response.noContent().build();
-    } catch (ObjectNotFoundException e) {
-      LOG.error("Error when trying to delete the attachment with id '{}' from entity with id '{}'", attachmentId, entityId, e);
-      return Response.status(Response.Status.NOT_FOUND).entity("AttachmentContext not found").build();
+    Identity userIdentity = getCurrentUserIdentity();
+    Boolean canDelete = attachmentService.canDelete(userIdentity, entityType, entityId);
+    if (canDelete) {
+      try {
+        attachmentService.deleteAttachmentItemById(entityId, entityType, attachmentId);
```

Comment:
```suggestion
        attachmentService.deleteAttachmentItemById(userIdentityId, entityId, entityType, attachmentId);
```
The user identity identifier has to been passed to Service layer to let it to do the fine ACL checks in this layer instead of embedding them in REST layer. Reminder: no business logic in REST layer.